Assets/Scripts/Imported/Lobby/MainLobbyScreen.cs

```diff
@@ -13,7 +13,11 @@ public class MainLobbyScreen : MonoBehaviour {
     private float _nextRefreshTime;
 
     private void Update() {
-        if (Time.time >= _nextRefreshTime) FetchLobbies();
+        if (Time.time >= _nextRefreshTime)
+        {
+            _nextRefreshTime = Time.time + _lobbyRefreshRate;
```

Comment:
`async void` functions return before their contents are ran. I noticed I was getting rate limited, I think there was a pretty big race condition here. Moving the time calculation into before the function is called enforces the rate limiter.